keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,790 @@
+- Multiple runtime profiles has never been requested as a feature.
+
+If built-in support for multiple runtime profiles is needed in the future, a new
+KEP will be created to cover its details.
+
+## Design Details
+
+### Failure and Fallback Strategy
+
+There are different scenarios in which applying an AppArmor profile may fail,
+below are the ones we mapped and their outcome once this KEP is implemented:
+
+| Scenario                                                                                           | API Server Result    | Kubelet Result                                                                                                                                        |
+| -------------------------------------------------------------------------------------------------- | -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 1) Using custom or `runtime/default` profile when container runtime does not support AppArmor.     | Pod created          | The outcome is container runtime dependent. In this scenario containers may 1) fail to start or 2) run normally without having its policies enforced. |
+| 2) Using custom or `runtime/default` profile that restricts actions a container is trying to make. | Pod created          | The outcome is workload and AppArmor dependent. In this scenario containers may 1) fail to start, 2) misbehave or 3) log violations.                  |
+| 3) Using custom profile that does not exist on the node.                                           | Pod created          | Containers fail to start. Retry respecting RestartPolicy and back-off delay.                                                                          |
+| 4) Using an unsupported runtime profile (i.e. `runtime/default-audit`).                            | Pod **not** created. | N/A                                                                                                                                                   |
+| 5) Using localhost profile with invalid name                                                       | Pod **not** created. | N/A                                                                                                                                                   |
+| 6) AppArmor is disabled by the host or the build                                                   | Pod created. | Kubelet puts Pod in blocked state.                                                                                                                    |
+
+Scenario 2 is the expected behavior of using AppArmor and it is included here
+for completeness.
+
+Scenario 5 represents the case of failing the existing validation, which is
+defined at [Pod API](#pod-api).
+
+### Version Skew Strategy
+
+All API skew is resolved in the API server. New Kubelets will only use the
```

Comment:
What happens if I have a new kube-apiserver and old kubelets?  this is common during upgrades and kubelets two levels back are allowed.

What happens if a new kubelet speaks to an older kube-apiserver. Say during a downgrade.

Reply:
This is called out in the [Upgrade / Downgrade](https://github.com/kubernetes/enhancements/pull/3298/files#diff-489f1a6d2c37c26084f3acf91981c36482a74fc859ffd0d31621be82a445d2ffR352-R355) section I believe.  Would it be better suited here?

>Specifically, annotation support will be removed
in the kubelet after this period, and fields will no longer be copied to annotations for older kubelet 
versions. However, annotations submitted to the API server will continue to be copied to fields at the 
kubelet indefinitely, as was done with Seccomp.